apps/frontend/src/app/3_organisms/ConversionsHistoryFrame/components/AmmConversionsHistoryFrame/AmmConversionsHistoryFrame.tsx

```diff
@@ -0,0 +1,169 @@
+import React, {
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { t } from 'i18next';
+import { nanoid } from 'nanoid';
+
+import {
+  Table,
+  Pagination,
+  OrderOptions,
+  OrderDirection,
+  NotificationType,
+  ErrorBadge,
+  ErrorLevel,
+} from '@sovryn/ui';
+
+import { ExportCSV } from '../../../../2_molecules/ExportCSV/ExportCSV';
+import {
+  DEFAULT_HISTORY_FRAME_PAGE_SIZE,
+  EXPORT_RECORD_LIMIT,
+} from '../../../../../constants/general';
+import { useNotificationContext } from '../../../../../contexts/NotificationContext';
+import { useAccount } from '../../../../../hooks/useAccount';
+import { useBlockNumber } from '../../../../../hooks/useBlockNumber';
+import { useMaintenance } from '../../../../../hooks/useMaintenance';
+import { translations } from '../../../../../locales/i18n';
+import { rskClient } from '../../../../../utils/clients';
+import {
+  Swap,
+  useGetSwapHistoryLazyQuery,
+} from '../../../../../utils/graphql/rsk/generated';
+import { dateFormat } from '../../../../../utils/helpers';
+import {
+  columnsConfig,
+  generateRowTitle,
+} from './AmmConversionsHistoryFrame.utils';
+import { useGetAMMConversionsHistory } from './hooks/useGetAMMConversionsHistory';
+
+const pageSize = DEFAULT_HISTORY_FRAME_PAGE_SIZE;
+
+export const AmmConversionsHistoryFrame: React.FC<PropsWithChildren> = ({
```

Comment:
All conversions history frames components are almost identical. They just differ in a few things. Extract all the common functionality and the render method to `BaseConversionsHistoryFrame` and pass the different things via props/util methods/custom hooks.